F50/Preferences/Preferences.shared.cs

```diff
@@ -10,10 +10,29 @@ namespace Xamarin.F50
     /// </summary>
     public partial class Preferences
     {
-        public static void DoStuff()
-        {
+		public Preferences()
+		{
+		}
 
-        }
-    }
+		public Preferences(string sharedName)
+		{
+			SharedName = sharedName;
+		}
+
+		public string SharedName { get; private set; }
+
+		public string Get(string key, string defaultValue)

```

Comment:
XML comments?

Reply:
We are going to be doing docs using mdoc.

Reply:
Ahhh... does that auto do it?

Reply:
Also does that get put into intellisense like xml docs?

Reply:
I don't think so as it is a separate task.

Reply:
I think we need to think about this, mdoc is nice for separating docs from the code, but it then needs to be put back into the nuget... This is going to be extra work.

Reply:
The advantage of mdoc is that we don't clutter the code with a 20-line snippet with weird xml escaping in the comment above a method or class. Also, we may want to write some awesome docs at the class, and this may take up space. And, if it is XML, contributors may feel more at east than modifying code to change a doc.

Reply:
We could do like Xamarin.Forms and have a separate step that creates the package. I do not fear the .nuspec and think we can do better than the automation.

For example this is in the nuget:
```
<file src="...\bin\Release\netstandard1.0\Xamarin.F50.dll" target="lib\netstandard1.0\Xamarin.F50.dll" />
```
that should actually be (note the `ref` instead of `lib`):
```
<file src="...\bin\Release\netstandard1.0\Xamarin.F50.dll" target="ref\netstandard1.0\Xamarin.F50.dll" />
```

Reply:
We need to support the case where the API is slightly different:

Android:
```csharp
class Platform {
    Init(Context);
}
```
iOS:
```csharp
class Platform {
    Init();
}
```

Reply:
I just tested the above case, the `--preserve` flag does not remove members and basically forms a union of the APIs

Reply:
I created this PR to add mdoc: https://github.com/xamarin/f50/pull/3

Reply:
Do you have an example of how we package up mdoc today in other nugets? It should be completely possible to do this

Reply:
I do it "manually" for SkiaSharp. I use a nuspec and add the xml files to each lib directory in the nuspec.